### PR TITLE
4314: Local DB scripts 

### PR DESCRIPTION
**Pull request by assoomro**

### CODE REVIEW

File: Start Local Db.ps1 [hunk 1]
```diff
@@ -0,0 +1,65 @@
+﻿param (
+    [int]$port = 1430,
+    [string]$username = "NHSD",
+    [string]$password = "Str0nk3RP@55W0rD"
```
Comment (sam-roberts-bjss) [inline Start Local Db.ps1]: Please could we change the account settings to match the other docker files?

File: Start Local Db.ps1 [hunk 2]
```diff
@@ -0,0 +1,65 @@
+﻿param (
```
Comment (sam-roberts-bjss) [inline Start Local Db.ps1]: Could we rename the file to Create Local Db?

File: src/NHSD.BuyingCatalogue.API/Infrastructure/Settings.cs [hunk 3]
```diff
@@ -10,7 +9,7 @@ internal class Settings : ISettings
         private readonly IConfiguration _configuration;
 
         public Settings(IConfiguration configuration) => _configuration = configuration.ThrowIfNull();
-
-        public string ConnectionString => Environment.GetEnvironmentVariable("ConnectionStrings__BuyingCatalogue") ??  _configuration["ConnectionStrings__BuyingCatalogue"];
+        
+        public string ConnectionString => _configuration.GetConnectionString("BuyingCatalogue");
```
Comment (sam-roberts-bjss) [inline src/NHSD.BuyingCatalogue.API/Infrastructure/Settings.cs]: Can we check for unit tests on this?

File: Start Local Db.ps1 [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+docker-compose -f "docker/docker-compose.debug.yml" down -v --rmi "all"
+
+validate 
+
+setEnvVariables
+
+docker-compose -f "docker/docker-compose.debug.yml" up -d
+
+$connectionString = "Data Source=127.0.0.1,$port;Initial Catalog=buyingcataloguegpit;MultipleActiveResultSets=True;User Id=$username;Password=$password"
```
Comment (sam-roberts-bjss) [inline Start Local Db.ps1]: The name of the database is mentioned in a couple of places. Could this be put into a variable?

File: docker/docker-compose.debug.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
+version: '3.5'
+
+services:
+
+  db:
+    image: nhsd-buying-catalogue-debug-db:latest
```
Comment (sam-roberts-bjss) [inline docker/docker-compose.debug.yml]: The image in the development docker file uses slashes. Can we follow that pattern?

File: Start Local Db.ps1 (hunk 4 above)
Comment (assoomro) [reply Start Local Db.ps1]: 👍

File: docker/docker-compose.debug.yml (hunk 5 above)
Comment (assoomro) [reply docker/docker-compose.debug.yml]: 👍

File: src/NHSD.BuyingCatalogue.API/Infrastructure/Settings.cs (hunk 3 above)
Comment (assoomro) [reply src/NHSD.BuyingCatalogue.API/Infrastructure/Settings.cs]: 👍

File: Start Local Db.ps1 (hunk 2 above)
Comment (assoomro) [reply Start Local Db.ps1]: 👍

File: Start Local Db.ps1 (hunk 1 above)
Comment (assoomro) [reply Start Local Db.ps1]: 👍

File: README.md [hunk 6]
```diff
@@ -103,6 +104,40 @@ bash tear_down_environment.sh
...
+```
+& '.\Create Local Db.ps1'
+```
+
+### On Linux/Mac box
+```
+bash create_local_db.sh
+```
+
+Secondly, copy and paste the connection string into your User Secrets file (in Visual Studio: src -> Right click on NHSD.BuyingCatalogue.API -> Manage User Secrets )
```
Comment (sam-roberts-bjss) [inline README.md]: Not everyone is using visual studio. I wonder whether we can use a link instead to refer the user to the official document regarding user secrets. It has platform specific instructions on there as well.

https://docs.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-3.1&tabs=windows

If we do this, then we can move this text under then "Local Debugging" heading as a summary.

**Comment by sam-roberts-bjss** [comment]
Also the PR indicates the branch is out of date.

File: README.md (hunk 6 above)
Comment (assoomro) [reply README.md]: Updated the README.md according to our conversation

**Pull request merged**